src/util/memory_manager.cpp

```diff
@@ -13,19 +13,17 @@ Copyright (c) 2015 Microsoft Corporation
 #include "util/error_codes.h"
 #include "util/debug.h"
 #include "util/scoped_timer.h"
-#ifdef __GLIBC__
+#ifndef HAS_MALLOC_USABLE_SIZE
+// drop calls to malloc_usable_size
+#elif defined(__GLIBC__)
 # include <malloc.h>
-# define HAS_MALLOC_USABLE_SIZE
```

Comment:
Not sure about this: the assumption of what has to go into the build configuration changes materially.

Reply:
Not exactly sure what you mean, but I pushed another change. Does that look better?

Reply:
Do you mean "whether to call `malloc_usable_size()` or not" is too low-level to be added to the build configuration?

Reply:
You are changing the build behavior. 
The behavior before: user doesn't configure anything. For selected configurations it uses the usable size.
New behavior: the user has to explicitly configure, with the CMAKE build only, the HAS_MALLOC_USABLE_SIZE. Otherwise it reverts to not using the usable size. 

Even this new code:
```
#ifndef USE_MALLOC_USABLE_SIZE
// drop calls to malloc_usable_size
#elif defined(__GLIBC__)
# include <malloc.h>
# define HAS_MALLOC_USABLE_SIZE
#elif defined(__APPLE__)
# include <malloc/malloc.h>
# define HAS_MALLOC_USABLE_SIZE
# define malloc_usable_size malloc_size
#elif defined(__FreeBSD__)
# include <malloc_np.h>
# define HAS_MALLOC_USABLE_SIZE
#elif defined(_WINDOWS)
# include <malloc.h>
# define HAS_MALLOC_USABLE_SIZE
# define malloc_usable_size _msize
#endif
```

doesn't make sense to me:

#ifndef USE_MALLOC_USABLE_SIZE
 // use malloc usable size is not defined
#else
 // use malloc usable size is defined. The uses of #define HAS_MALLOC_USABLE_SIZE are redundant.
 
 Is this really what you intend? Do you intend #ifdef instead of #ifndef?
 I just can't take pull requests that don't make sense. In the end I have to fumble with regressions.

Reply:
I see. I pushed [another change](https://github.com/Z3Prover/z3/compare/master...remysucre:z3:master) to keep `malloc_usable_size` by default, and the user can pass an `Z3_AVOID_MALLOC_USABLE_SIZE` option to drop the calls. I believe this implements exactly the recommendation [here](https://github.com/Z3Prover/z3/issues/6890#issuecomment-1723508107) which got a 👍 from you some time ago. Should I open another PR?